Protocol.md

```diff
@@ -23,188 +44,182 @@ The following figure gives a brief overview of the protocol:
-The following public keys:
+The receiving node MAY reject the contract if:
 
-1. Funding Public Key
-   - B's public key used in the multi-signature [funding output](Transactions.md#FundingOutputs)
-2. Sweep Public Key
-   - B's public key used in the creation of the CETs [output script](Transactions.md#CETOutputs).
+  - it does not agree to the terms in `contract_info`.
+  - the `contract_info` is missing relevant events.
+  - it does not want to use the oracle(s) specified in `oracle_info`.
+  - `total_collateral_satoshis` is too small.
 
-#### B's addresses
+The receiving node MUST reject the contract if:
 
-The following addresses:
+  - the `chain_hash` value is set to a hash of a chain that is unknown to the receiver.
+  - the `contract_info` refers to events unknown to the receiver.
+  - the `oracle_info` refers to an oracle unknown to the receiver.
+  - it considers `feerate_per_vb` too small for timely processing or unreasonably large.
+  - `funding_pubkey` is not a valid secp256k1 pubkey in compressed format.
+  - the funder's amount for the funding transaction is not sufficient for full [fee payment](Transactions.md#fee-payment).
 
-1. Funding Change Address
-   - The address to use to send the change if the sum of A's inputs is greater than the fund amount plus the fees.
-2. Final Address
-   - The address to which funds will be sent in [unilateral contract execution](Transactions.md#CETOutputs), [refund](Transactions.md#refund-transaction) and [mutual closing](Transactions.md#mutual-closing-transaction) transactions.
+### The `accept_dlc` Message
 
-#### B's inputs
-The set of UTXOs to be used as input to the fund transaction.
+This message contains information about a node and indicates its
+acceptance of the new DLC, as well as its CET and refund transaction
+signatures. This is the second step toward creating the funding transaction
+and closing transactions.
 
-#### CET signatures
-A set of signatures from B, one for each CET input.
+1. type: ??? (`accept_dlc`)
+2. data:
+   * [`32*byte`:`temporary_contract_id`]
+   * [`u64`:`total_collateral_satoshis`]
+   * [`point`:`funding_pubkey`]
+   * [`spk`:`payout_spk`]
+   * [`u16`:`num_funding_inputs`]
+   * [`num_funding_inputs*funding_input`:`funding_inputs`]
+   * [`spk`:`change_spk`]
+   * [`cet_signatures`:`cet_signatures`]
+   * [`signature`:`refund_signature`]
 
-#### Refund signature
-A signature from B for the refund transaction input.
+#### Requirements
 
----
+The `temporary_contract_id` MUST be the SHA256 hash of the `offer_dlc` message.
 
-### Sign
+The sender MUST:
 
-After receiving and [validating](#sign-validation) the content of the `offer` message, A replies with a `sign` message containing:
-1. Fund transaction signatures
-1. CET signatures
-1. Refund signature
+  - set `total_collateral_satoshis` sufficiently large so that the sum of both parties' total collaterals is at least as large as the largest payout in the `offer_dlc`'s `contract_info`.
```

Comment:
Again I'm quite confused by this. I feel the required collateral should be clear from the offer message, I don't really see in which case you would want it to be greater?